### PR TITLE
Tr3 tribesmen fix, Dart_Emitter rotation to every angle 

### PR DESCRIPTION
**Pull request by Nemoel-Tomo**
Tr3 Tribesmen with axe now will attack Lara,
Tr3 Tribesmen with dart now shoots correctly at Lara,
Dart Emitter can now be rotated to every angle
Dart now draws real mesh from ID_DARTS slot
You can also set the damage value for Dart-Emitter with ocb. Negative numbers means that the darts will poison Lara in addition. OCB 0 means default damage (25)

### CODE REVIEW

File: TombEngine/Objects/Generic/Traps/dart_emitter.cpp [hunk 1]
```diff
@@ -19,13 +19,16 @@ namespace TEN::Entities::Traps
 		{
 			DoDamage(LaraItem, 25);
 			DoBloodSplat(item->Pose.Position.x, item->Pose.Position.y, item->Pose.Position.z, (GetRandomControl() & 3) + 4, LaraItem->Pose.Orientation.y, LaraItem->RoomNumber);
+			
+			if (item->TriggerFlags)
 			Lara.PoisonPotency += 1; // Was 160 with the total poison potency later shifted right by 8 when applied to Lara's health. The effect was that each dart contributed a mere fraction to the potency. @Sezz 2022.03.09
```
Comment (Lwmte) [inline TombEngine/Objects/Generic/Traps/dart_emitter.cpp:24]: I'd also customize the amount of damage and poisoning that darts can inflict. For example, by default you can set damage to 25 (by the way, you can make it constexpr DEFAULT_DART_DAMAGE), but if OCB is something else, then use this value.

Negative values may mean that darts only decrease specified damage from Lara, while positive values may also add poison to her. For example, dart emitter with OCB 120 will decrease health by 120 and also poison her, while OCB -120 may just decrease her health.

File: TombEngine/Objects/Generic/Traps/dart_emitter.cpp [hunk 2]
```diff
@@ -76,65 +79,21 @@ namespace TEN::Entities::Traps
...
-			case -0x4000:
-				x = -WALL_SIZE / 2;
-				break;
-			}
-
-			dartItem->Pose.Position.x = x + item->Pose.Position.x;
-			dartItem->Pose.Position.y = item->Pose.Position.y - WALL_SIZE / 2;
-			dartItem->Pose.Position.z = z + item->Pose.Position.z;
+			dartItem->Pose.Position.x = item->Pose.Position.x ;
+			dartItem->Pose.Position.y = item->Pose.Position.y -CLICK(0.9);
```
Comment (Lwmte) [inline TombEngine/Objects/Generic/Traps/dart_emitter.cpp:83]: Is there a space missing before CLICK?

File: TombEngine/Objects/Generic/Traps/dart_emitter.cpp [hunk 3]
```diff
@@ -76,65 +79,21 @@ namespace TEN::Entities::Traps
...
+			dartItem->Pose.Position.z = item->Pose.Position.z;
 
 			InitialiseItem(dartItemNumber);
 
-			dartItem->Pose.Orientation.x = 0;
-			dartItem->Pose.Orientation.y = item->Pose.Orientation.y + -ANGLE(180);
+			dartItem->Pose.Orientation.x = item->Pose.Orientation.x + -ANGLE(180);
```
Comment (Lwmte) [inline TombEngine/Objects/Generic/Traps/dart_emitter.cpp:88]: can this be simplified to just `item->Pose.Orientation.y - ANGLE(180)`?

File: TombEngine/Objects/TR3/Entity/tr3_tribesman.cpp [hunk 4]
```diff
@@ -385,6 +386,7 @@ namespace TEN::Entities::Creatures::TR3
 		short tilt = 0;
 		auto extraHeadRot = EulerAngles::Zero;
 		auto extraTorsoRot = EulerAngles::Zero;
+		short torso = 0;
```
Comment (Lwmte) [inline TombEngine/Objects/TR3/Entity/tr3_tribesman.cpp:389]: What is the purpose of this variable?

File: TombEngine/Objects/Generic/Traps/dart_emitter.cpp (hunk 1 above)
Comment (Nemoel-Tomo) [reply TombEngine/Objects/Generic/Traps/dart_emitter.cpp:24]: That's a good idea. I made negative numbers to poison Lara and positive numbers for non-poison damage

File: TombEngine/Objects/Generic/Traps/dart_emitter.cpp (hunk 2 above)
Comment (Nemoel-Tomo) [reply TombEngine/Objects/Generic/Traps/dart_emitter.cpp:83]: ah, fixed :)

File: TombEngine/Objects/Generic/Traps/dart_emitter.cpp (hunk 3 above)
Comment (Nemoel-Tomo) [reply TombEngine/Objects/Generic/Traps/dart_emitter.cpp:88]: fixed

File: TombEngine/Objects/TR3/Entity/tr3_tribesman.cpp (hunk 4 above)
Comment (Nemoel-Tomo) [reply TombEngine/Objects/TR3/Entity/tr3_tribesman.cpp:389]: oh, this was a remnant of one of my tests. I removed it.

File: TombEngine/Objects/Generic/Traps/dart_emitter.cpp [hunk 5]
```diff
@@ -11,21 +11,29 @@
...
+				DoDamage(LaraItem, abs(item->TriggerFlags));
+
 			DoBloodSplat(item->Pose.Position.x, item->Pose.Position.y, item->Pose.Position.z, (GetRandomControl() & 3) + 4, LaraItem->Pose.Orientation.y, LaraItem->RoomNumber);
-			Lara.PoisonPotency += 1; // Was 160 with the total poison potency later shifted right by 8 when applied to Lara's health. The effect was that each dart contributed a mere fraction to the potency. @Sezz 2022.03.09
 			KillItem(itemNumber);
 		}
 		else
 		{
 			int oldX = item->Pose.Position.x;
-			int oldZ = item->Pose.Position.z - 1000;
```
Comment (Lwmte) [inline TombEngine/Objects/Generic/Traps/dart_emitter.cpp:28]: What is this value of 1000 and why it was removed?

**Pull request merged**